Makefile: fix GEOS uptodate dependency

otan opened the pull request:
When building GEOS, ensure that the binaries/libs in /go/native are up
to date instead of the local lib directory. Otherwise, even though the
local ./lib/*.so files may be up to date, it's dependency may not be.
e.g. in the case of x86_64-pc-linux-gnu and x86_64-pc-unknown-gnu,
though the so file may exist for x86_64-pc-unknown-gnu in
lib/libgeos_c.so, and so uptodate thinks it's correct, however, the file
may not be built in /go/native/x86_64-pc-unknown-gnu/lib/libgeos_c.so,
and so the copy fails as now file exists there.

Also added a check in compile-builds for this behavior.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50926)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50926#-:-MBFPPR76pxUMmZGaNm2:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan committed:
Makefile: fix GEOS uptodate dependency

When building GEOS, ensure that the binaries/libs in /go/native are up
to date instead of the local lib directory. Otherwise, even though the
local ./lib/*.so files may be up to date, it's dependency may not be.
e.g. in the case of x86_64-pc-linux-gnu and x86_64-pc-unknown-gnu,
though the so file may exist for x86_64-pc-unknown-gnu in
lib/libgeos_c.so, and so uptodate thinks it's correct, however, the file
may not be built in /go/native/x86_64-pc-unknown-gnu/lib/libgeos_c.so,
and so the copy fails as now file exists there.

Also added a check in compile-builds for this behavior.

Release note: None

otan commented:
bleeding edge: https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_MergeToMaster&branch_Cockroach=50926&tab=buildTypeStatusDiv

otan commented:
this works!

bors r=petermattis
